### PR TITLE
docs: Add CITATION.cff file for citation metadata

### PR DESCRIPTION
**Pull request by matthewfeickert**
* Add CITATION.cff to provide machine and human readable citation metadata.

* Remove .zenodo.json metadata file as reundant to CITATION.cff as Zenodo will recognize and read CITATION.cff and CITATION.cff provides more functionality.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,63 @@
+cff-version: 1.2.0
+message: "Please cite the following works when using this software."
+type: software
+authors:
+- family-names: "Guest"
+  given-names: "Daniel"
+  orcid: "https://orcid.org/0000-0002-4305-2295"
+  affiliation: "Humboldt-Universität zu Berlin"
```
Comment (matthewfeickert) [inline CITATION.cff:8]: I've updated affiliation information from the original metadata to I believe the locations people were at for the latest release.

**Review by matthewfeickert — COMMENTED** [review]
Some high level notes for reviewers.

**Comment by matthewfeickert** [comment]
@dguest this is ready for review. To preview this if you open my fork on the PR branch then you'll get the "Cite this repository" button on the right hand side of GitHub's UI.

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  given-names: "Benjamin"
+  orcid: "https://orcid.org/0000-0002-3163-1062"
+  affiliation: "Universität Regensburg"
+
+- family-names: "Feickert"
+  given-names: "Matthew"
+  orcid: "https://orcid.org/0000-0003-4124-7862"
+  affiliation: "University of Wisconsin-Madison"
+title: "lwtnn: v2.13"
+version: 2.13
```
Comment (matthewfeickert) [inline CITATION.cff:55]: Version metadata needs to be updated before or on release tag commits.

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+doi: 10.5281/zenodo.597221
+repository-code: "https://github.com/lwtnn/lwtnn/releases/tag/v2.13"
+url: "https://github.com/lwtnn/lwtnn/"
+keywords:
```
Comment (dguest) [inline CITATION.cff:59]: I was wondering if there was some kind of "inference"-like keyword, but looking at similar packages like [onnxruntime](https://github.com/microsoft/onnxruntime/blob/main/CITATION.cff) or [frugally-deep](https://github.com/Dobiasd/frugally-deep/blob/master/CITATION.cff) there seems to be no consistency anyway. Are keywords always a bag of entropy or is there some way to bring order to this mess?

Anyway, that's just a thought for later, I think what you've done here is amazing so thanks!

File: CITATION.cff (hunk 1 above)
Comment (dguest) [reply CITATION.cff:8]: It's tempting to remove affiliation completely for people with a valid orcid (affiliation seems like a relic from before unique identifiers existed, which is outdated half the time for most active researchers), but I wouldn't do that without asking everyone anyway and I don't know the wider practices here. Again, just a discussion for later, not important here.

**Review by dguest — APPROVED** [review]

File: CITATION.cff (hunk 3 above)
Comment (matthewfeickert) [reply CITATION.cff:59]: > Are keywords always a bag of entropy

@dguest Basically bag of entropy. For my own projects I just think about what describes them best in my own mind and then slap them down there.

**Pull request merged**